app/src/main/java/com/swent/suddenbump/model/user/UserRepositoryFirestore.kt

```diff
@@ -36,7 +37,8 @@ import kotlinx.coroutines.flow.MutableStateFlow
 class UserRepositoryFirestore(
     private val db: FirebaseFirestore,
     private val sharedPreferencesManager: SharedPreferencesManager,
-    private val workerScheduler: WorkerScheduler
+    private val workerScheduler: WorkerScheduler,
+    private val context: Context
```

Comment:
bad practice. view comment below on how to handle this better to avoid memory leaks.